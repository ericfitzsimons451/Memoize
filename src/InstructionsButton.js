import React from 'react';
import './main.scss'

function InstructionsButton(props) {
  return(
    <div className="instructions-style-div">
      <button className="instructions-button" type="button" onClick={props.displayInstructions}>Click for Instructions</button>
    </div>
  )
}



export default InstructionsButton;