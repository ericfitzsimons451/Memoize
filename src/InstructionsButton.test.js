import React from 'react';
import InstructionsButton from './InstructionsButton.js';
import { shallow } from 'enzyme';

const mockDisplayInstructions = jest.fn();

describe('InstructionsButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <InstructionsButton displayInstructions={mockDisplayInstructions} />
      )
  })

  it('should match the snapshot with all the correct data passed to the InstructionsButton', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should register a click when CLICK FOR INSTRUCTIONS button has been triggered', () => {
    wrapper.find('.instructions-button').simulate('click')
    expect(mockDisplayInstructions).toBeCalled();
  })

})