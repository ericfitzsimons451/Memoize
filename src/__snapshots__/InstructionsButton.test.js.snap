// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionsButton should match the snapshot with all the correct data passed to the InstructionsButton 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InstructionsButton
    displayInstructions={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="instructions-button"
        onClick={[MockFunction]}
        type="button"
      >
        Click for Instructions
      </button>,
      "className": "instructions-style-div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Click for Instructions",
        "className": "instructions-button",
        "onClick": [MockFunction],
        "type": "button",
      },
      "ref": null,
      "rendered": "Click for Instructions",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="instructions-button"
          onClick={[MockFunction]}
          type="button"
        >
          Click for Instructions
        </button>,
        "className": "instructions-style-div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click for Instructions",
          "className": "instructions-button",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "Click for Instructions",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
